### PR TITLE
feat: make Layout setting cumulative

### PR DESCRIPTION
**Pull request by brillout**
Main commit: https://github.com/vikejs/vike-vue/commit/346981c01b99d13ea1ef08bafa2d4630918585cb. (The other commits merely do some refactoring and add a nested layout example.)

@pdanpdan Do the changes to `createVueApp.ts` look good to you? I did some refactoring. Btw. I've been wondering: how about we use `shallowRef()` instead of `ref` inside `createVueApp.ts`?

### CODE REVIEW

File: examples/full/pages/starship/+Layout.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  <p>
+    <b>
+      State is preserved upon navigating within the nested layout. Click on the counter then on the links below:
+      observe how the counter state is preserved.
+    </b>
+    <Counter />
+  </p>
+  <br />
+  <div class="sub-navigation">
+    <a keep-scroll-position href="/starship">Overview</a> <a keep-scroll-position href="/starship/reviews">Reviews</a>
```
Comment (pdanpdan) [inline examples/full/pages/starship/+Layout.vue:18]: I would split this in 2 lines

File: packages/vike-vue/src/renderer/createVueApp.ts [hunk 2]
```diff
@@ -10,22 +10,25 @@ import { isPlainObject } from '../utils/isPlainObject'
 import { setData } from '../hooks/useData'
 
 type ChangePage = (pageContext: PageContext) => Promise<void>
-async function createVueApp(pageContext: PageContext, ssr: boolean, rootComponentName: 'Head' | 'Page') {
-  const rootComponentRef = ref(markRaw(pageContext.config[rootComponentName]))
-  const layoutRef = ref(markRaw(pageContext.config.Layout))
+async function createVueApp(pageContext: PageContext, ssr: boolean, mainComponentName: 'Head' | 'Page') {
+  const mainComponentRef = ref(markRaw(pageContext.config[mainComponentName]))
```
Comment (pdanpdan) [inline packages/vike-vue/src/renderer/createVueApp.ts:14]: These 2 can be `shallowRef` as far as I'm concerned, but I think `ref(markRaw(x))` does the same thing - I'm not very sure, I have to check

File: packages/vike-vue/src/renderer/createVueApp.ts [hunk 3]
```diff
@@ -43,8 +46,8 @@ async function createVueApp(pageContext: PageContext, ssr: boolean, rootComponen
     assertDataIsObject(data)
     objectReplace(dataReactive, data)
     objectReplace(pageContextReactive, pageContext)
-    rootComponentRef.value = markRaw(pageContext.config[rootComponentName])
-    layoutRef.value = markRaw(pageContext.config.Layout)
+    mainComponentRef.value = markRaw(pageContext.config[mainComponentName])
```
Comment (pdanpdan) [inline packages/vike-vue/src/renderer/createVueApp.ts:49]: If we change to `shallowRef` above here we don't need `markRaw`

**Comment by pdanpdan** [comment]
I would go with a merge for this one (because it works as expected) and after that start tweaking

**Comment by brillout** [comment]
> These 2 can be `shallowRef` as far as I'm concerned, but I think `ref(markRaw(x))` does the same thing - I'm not very sure, I have to check

:+1: Let me know what you find and PR welcome.

**Comment by brillout** [comment]
Let's release after we merge `clientOnly()`.

**Comment by brillout** [comment]
Released in `vike-vue@0.7.3`. I decided to release it as a minor despite the breaking chagne; let me know if you disagree with that decision.

**Pull request merged**